Library/Homebrew/cmd/update.sh

```diff
@@ -302,6 +312,9 @@ EOS
   # kill all of subprocess on interrupt
   trap '{ pkill -P $$; wait; exit 130; }' SIGINT
 
+  HOMEBREW_UPDATE_FAILED_FILE="$HOMEBREW_REPOSITORY/.git/UPDATE_FAILED"
+  rm -rf "$HOMEBREW_UPDATE_FAILED_FILE"
```

Comment:
`-f` instead of `-rf` should be enough to remove this file (and less dangerous).